docs/influence/index.md

```diff
@@ -249,6 +249,87 @@ not to corrupt the outcome too much, the parameter $\lambda$ should be as small
 as possible while still allowing a reliable inversion of $H_{\hat{\theta}} +
 \lambda \mathbb{I}$.
 
+### Block-diagonal approximation
+
+This implementation is capable of using a block-diagonal approximation.
+The full matrix is approximated by a block-diagonal version, which
+reduces both the time and memory consumption.
+The blocking structure can be specified via the `block_structure` parameter.
+The `block_structure` parameter can either be a
+[BlockMode][pydvl.influence.torch.util.BlockMode] enum (which provides
+layer-wise or parameter-wise blocking) or a custom block structure defined
+by an ordered dictionary with the keys being the block identifiers (arbitrary
+strings) and the values being lists of parameter names contained in the block.
+```python
+from torch.utils.data import DataLoader
+from pydvl.influence.torch import DirectInfluence, BlockMode, SecondOrderMode
+
+training_data_loader = DataLoader(...)
+# layer-wise block-diagonal approximation
+infl_model = DirectInfluence(model, loss,
+                             regularization=0.1,
+                             block_structure=BlockMode.LAYER_WISE)
+
+block_structure = OrderedDict((
+    ("custom_block1", ["0.weight", "1.bias"]), 
+    ("custom_block2", ["1.weight", "0.bias"]),
+))
+# custom block-diagonal structure
+infl_model = DirectInfluence(model, loss,
+                             regularization=0.1,
+                             block_structure=block_structure)
+infl_model = infl_model.fit(training_data_loader)
+```
+If you would like to apply a block-specific regularization, you can provide a
+dictionary with the block names as keys and the regularization values as values.
+In this case, the specification must be complete, i.e. every block must have
```

Comment:
We could also fill everything for which no regularization was provided with a default value (probably 0)

Reply:
This is the actual behavior, but the documentation is wrong, as it describes a special case.